### ISSUE TITLE
Boolean comparison vs tuple fails in 0.17.0

### ISSUE THREAD

**Issue by svidela** [pydata/pandas#11339]
Just updated to 0.17.0 (python 2.7.9 on OS X Yosemite) and I found the following:

```
>>> import pandas as pd
>>> pd.__version__
u'0.17.0'
>>> s = pd.Series([(1,1),(1,2)])
>>> s==(1,2)
0    False
1    False
dtype: bool
```

Meanwhile in 0.16.2:
```
>>> import pandas as pd
>>> pd.__version__
'0.16.2'
>>> s = pd.Series([(1,1),(1,2)])
>>> s==(1,2)
0    False
1     True
dtype: bool
```

**Comment by shoyer** [comment]
Can you check that you're using the same version of numpy? I think this could actually be an issue with numpy 1.10 adjusting its casting rules...

**Comment by svidela** [comment]
Hi,

In both cases I have latest numpy (1.10.1). I don't know if it's related or not but I actually got to this issue because I have a Series where values are frozensets. In that case not only fails but there is also an exception. Maybe the traceback helps you to know if this is numpy- or just pandas-related

```
>>> s = pd.Series([frozenset([1]), frozenset([1,2])])
>>> s
0       (1)
1    (1, 2)
dtype: object
>>> s == frozenset([1,2])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-40-2d7a371a1a3d> in <module>()
----> 1 s == frozenset([1,2])

/Users/sancho/.virtualenvs/test/lib/python2.7/site-packages/pandas/core/ops.pyc in wrapper(self, other, axis)
    724                 other = np.asarray(other)
    725
--> 726             res = na_op(values, other)
    727             if isscalar(res):
    728                 raise TypeError('Could not compare %s type with Series'

/Users/sancho/.virtualenvs/test/lib/python2.7/site-packages/pandas/core/ops.pyc in na_op(x, y)
    644                     result = lib.vec_compare(x, y.astype(np.object_), op)
    645                 else:
--> 646                     result = lib.vec_compare(x, y, op)
    647             else:
    648                 result = lib.scalar_compare(x, y, op)

pandas/lib.pyx in pandas.lib.vec_compare (pandas/lib.c:14283)()

ValueError: Buffer has wrong number of dimensions (expected 1, got 0)
```

**Comment by jreback** [comment]
this was a bug introduced in #10569  I think. fixed in #11339

**Comment by svidela** [comment]
Should I create another issue for the exception when comparing versus a frozenset or it's already included in the same fix?

**Comment by jreback** [comment]
fixed up.. note that using ``object`` dtype things inside a Series is very inefficient.

**Comment by sinhrks** [comment]
The behavior may be changed after #13637. Pls comment if any thoughts.
